### ISSUE TITLE
GRPO train batchsize (7x1) not divided by 8

### ISSUE THREAD

**Issue by ZekaiGalaxy** [huggingface/open-r1#365]
Hi, I use 1 node of 8 H cards to run grpo example and encounter the following problem
<img width="1579" alt="Image" src="https://github.com/user-attachments/assets/40d96cc0-2672-4659-81a6-356bcf89b8eb" />

And here is my setting:
# Model arguments
model_name_or_path: XXX
model_revision: main
torch_dtype: bfloat16

# Data training arguments
dataset_name: XXX
dataset_configs:
- all
# Num processes is less by 1 as vLLM is using 1 GPU
num_processes: 7

# GRPO trainer config
bf16: true
use_vllm: true
vllm_device: auto
vllm_gpu_memory_utilization: 0.7
do_eval: true
eval_strategy: steps
eval_steps: 100
gradient_accumulation_steps: 16
gradient_checkpointing: true
learning_rate: 2.0e-05
log_level: info
logging_steps: 10
logging_strategy: steps
lr_scheduler_type: cosine
max_prompt_length: 512
max_completion_length: 1024
max_steps: -1
num_train_epochs: 1
output_dir: data/Qwen2.5-1.5B-Open-R1-GRPO
overwrite_output_dir: true
per_device_eval_batch_size: 4   
per_device_train_batch_size: 1
report_to:
- wandb
save_strategy: "no"
seed: 42
warmup_ratio: 0.1

**Comment by qgallouedec** [comment]
You should try to set `num_generations` to 7 as suggested by the error message:

```python
GRPOConfig(..., num_generations=7)
```
